### PR TITLE
feat(apply): add integration flow that includes 3 cases [PHX-2769]

### PR DESCRIPTION
**Pull request by Vinz93**
## Integration test Apply command

The test creates an environment, and makes content changes to reproduce the main 3 cases (add, delete and update), then we run the create command to generate a `changeset.json`, then we run the apply command which uses the changset generated before.

Screenshot 2023-08-18 at 17 08 33

### Todo

- [ ] Clean up code
- [ ] Clean up types and helper functions

### CODE REVIEW

File: test/integration/commands/apply/index.test.ts [hunk 1]
```diff
@@ -1,52 +1,65 @@
...
-    cmaToken,
-    teardown,
-  }
-})
+  before(async () => {
+    const client = createClient({ accessToken: cmaToken })
+    testSpace = await client.getSpace(spaceId)
+    const environmentsContext = await createEnvironments(testSpace)
+    if (!environmentsContext) {
+      throw new Error('Environments were not created successfully')
```
Comment (ruderngespra) [inline test/integration/commands/apply/index.test.ts:28]: why do we need this, isn't an error thrown anyways if something goes wrong with the `testUtils` (https://github.com/contentful/contentful-merge/blob/feat%2Fintegration-tests-part-2/test/integration/commands/bootstrap.ts#L68

File: test/integration/commands/bootstrap.ts [hunk 2]
```diff
@@ -69,21 +61,31 @@ const teardown = async ({ apiKey, environments }: { apiKey?: ApiKey; environment
...
+    console.log('all setup')
+    return {
+      sourceEnvironment,
+      targetEnvironment,
+      cdaToken: apiKey.accessToken,
+      spaceId: testSpace.sys.id,
+      teardown: () => teardown({ apiKey, environments: [sourceEnvironment, targetEnvironment] }),
+    }
+  } catch (error) {
+    console.error(error)
```
Comment (ruderngespra) [inline test/integration/commands/bootstrap.ts:82]: why catch it here and then rethrow in the other file (see the other comment)?

File: test/integration/commands/create/index.test.ts [hunk 3]
```diff
@@ -4,41 +4,44 @@ import fs from 'fs'
...
+  const changesetPath = './changeset.json'
+  let testContext: TestContext
+  let testSpace: Space
+  let cdaTokenWithOnlyMasterAccess: ApiKey
+  before(async () => {
+    const client = createClient({ accessToken: cmaToken })
+    testSpace = await client.getSpace(spaceId)
+    const environmentsContext = await createEnvironments(testSpace)
+    if (!environmentsContext) {
+      throw new Error('Environments were not created successfully')
```
Comment (ruderngespra) [inline test/integration/commands/create/index.test.ts:26]: same as above

**Review by ruderngespra — APPROVED** [review]
lgtm, just a few comments / confusion but nothing blocking 

but I wonder if we should move the apply part to its own e2e folder as we are here now testing the whole flow instead of only the apply command

File: test/integration/commands/bootstrap.ts (hunk 2 above)
Comment (Vinz93) [reply test/integration/commands/bootstrap.ts:82]: Good point, I think it's like this for the debugging, I'll adjust it 👍

File: test/integration/commands/apply/index.test.ts (hunk 1 above)
Comment (Vinz93) [reply test/integration/commands/apply/index.test.ts]: I double-check and the reason of that check is typescript, `createEnvironments` returns `undefined` or result

**Pull request merged**